src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
```

Comment:
@davidsh, the WinHttp implementation isn't disposing of this X509Certificate2, and there's a WinHttp test that's then relying on that, expecting the certificate passed to the callback to be valid even after the callback returns.  That seems wrong, so I'm disposing this, but let me know if there's a good reason this shouldn't be disposed in the callback case.